[Onboarding] Additional auto-detect flow telemetry

mykolaharmash opened the pull request:
Closes https://github.com/elastic/observability-dev/issues/4236 🔒

This change:
* Moves flow progress update telemetry events to the backend (this will prevent situations when multiple progress updates happened in a short time interval, but frontend could only see the latest update because it checks for progress once in a few seconds which might not be enough in some situations)
* Leaves the data_ingested final event on the FE because it's not explicitly triggered on the BE
* Deletes a separate event type for auto detect flow in favor of the generic even used by other flows
* Adds progress updates to the auto-detect script when when it's being run on unsupported OS or lsof binary is not present
* Adds logging of the OS and architecture used by the host

Events coming from your local Kibana can viewed on the [🔒 Staging Telemetry cluster](https://telemetry-v2-staging.elastic.dev). Though keep in mind, there is a ~1 hour delay for events indexing 😔

Example event:
![CleanShot 2025-01-22 at 12 26 14@2x](https://github.com/user-attachments/assets/077e4f15-5283-4198-a543-30f2507fa0f5)


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

mykolaharmash commented on x-pack/solutions/observability/plugins/observability_onboarding/public/application/app.tsx:
```diff
@@ -53,10 +52,6 @@ export function ObservabilityOnboardingAppRoot({
 
-  core.analytics.reportEvent(OBSERVABILITY_ONBOARDING_TELEMETRY_EVENT.eventType, {
-    uses_legacy_onboarding_page: false,
-  });
```
Deleting this one as legacy onboarding page is not a thing anymore. Also deleted `uses_legacy_onboarding_page` property from the even schema.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/269315)
* Commit: 8023da9b32e439af58120a3df67e3e44f1cf0732

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f4ff699a82253b036f51367da2e09dc04483b60d) | [after](https://github.com/elastic/kibana/commit/8023da9b32e439af58120a3df67e3e44f1cf0732) | diff |
| --- | --- | --- | --- |
| `observabilityOnboarding` | 232 | 230 | -2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f4ff699a82253b036f51367da2e09dc04483b60d) | [after](https://github.com/elastic/kibana/commit/8023da9b32e439af58120a3df67e3e44f1cf0732) | diff |
| --- | --- | --- | --- |
| `observabilityOnboarding` | 269.7KB | 269.2KB | -521.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f4ff699a82253b036f51367da2e09dc04483b60d) | [after](https://github.com/elastic/kibana/commit/8023da9b32e439af58120a3df67e3e44f1cf0732) | diff |
| --- | --- | --- | --- |
| `observabilityOnboarding` | 10.2KB | 10.3KB | +127.0B |



<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/269315","number":269315,"commit":"8023da9b32e439af58120a3df67e3e44f1cf0732"}],"number":269315}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/270088)
* Commit: 776b7694681b84a4f5d6c0e4605e1b0ae81092e4
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-207726-776b7694681b`

### History
* :green_heart: [Build #269315](https://buildkite.com/elastic/kibana-pull-request/builds/269315) succeeded 8023da9b32e439af58120a3df67e3e44f1cf0732

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/270088","number":270088,"commit":"776b7694681b84a4f5d6c0e4605e1b0ae81092e4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/269315","number":269315,"commit":"8023da9b32e439af58120a3df67e3e44f1cf0732"}],"number":270088}
buildkite-pr-comment-->

flash1293 reviewed (commented):
I tested this for the full flow and it seems to work fine for the happy path (except for the "initialized" message, see below).

However it logs
> auto_detect.sh: line 16: update_step_progress: command not found

And when provoking a failure e.g. in the os/arch check, it just raises the same error again.

I got it to work by moving the `update_step_progress` definition further up, but then it fails with another error because the error message contains doublequotes which mess up the JSON of the payload.

So I think we need to do two things:
* Make sure the update_progress function is defined early enough
* Make sure the JSON payload is constructed correctly even for the error case (no doublequotes)

mykolaharmash commented:
Thank you @flash1293! The `initialize` event was a last minute addition and I did not re-test after adding it, sorry about this. As you've suggested, I moved the function definition up and removed the quotes from the error message.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/270374)
* Commit: 902446b1075ce4ec81f68537dca04d4bfa3eac6a
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-207726-902446b1075c`

### History
* :green_heart: [Build #270088](https://buildkite.com/elastic/kibana-pull-request/builds/270088) succeeded 776b7694681b84a4f5d6c0e4605e1b0ae81092e4
* :green_heart: [Build #269315](https://buildkite.com/elastic/kibana-pull-request/builds/269315) succeeded 8023da9b32e439af58120a3df67e3e44f1cf0732

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/270374","number":270374,"commit":"902446b1075ce4ec81f68537dca04d4bfa3eac6a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/270088","number":270088,"commit":"776b7694681b84a4f5d6c0e4605e1b0ae81092e4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/269315","number":269315,"commit":"8023da9b32e439af58120a3df67e3e44f1cf0732"}],"number":270374}
buildkite-pr-comment-->

flash1293 reviewed (approved):
Tested and works fine. 

However I'm still getting this misplaced warning in the script output:
<img width="632" alt="Screenshot 2025-01-27 at 17 18 40" src="https://github.com/user-attachments/assets/2f6583c1-e656-43be-8b69-85e095174311" />

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/12993396826

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.x|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 207726
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

mykolaharmash commented:
> Tested and works fine.
> 
> However I'm still getting this misplaced warning in the script output: <img alt="Screenshot 2025-01-27 at 17 18 40" width="632" src="https://private-user-images.githubusercontent.com/1508364/406998792-2f6583c1-e656-43be-8b69-85e095174311.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwNTM4MTAsIm5iZiI6MTczODA1MzUxMCwicGF0aCI6Ii8xNTA4MzY0LzQwNjk5ODc5Mi0yZjY1ODNjMS1lNjU2LTQzYmUtOGI2OS04NWUwOTUxNzQzMTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjhUMDgzODMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjViMGI4OTZmY2VjYzc4MGIwMzIwOWU3Y2IzYzQyMWRhMDhkOTFmZTI1YjQzNTU2NzYyOTJlMDczZTNiNDZjOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jclAxDX54sH7QXVR2yAsgx1hsa2IsNNji-O0rDqF6hk">

Seems like it was still the previous commit, it says line 16 but there is no function call on that line anymore

mykolaharmash commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/208489">](https://github.com/elastic/kibana/pull/208489)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.4 -->
